Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -4931,6 +4932,201 @@ RefPtr<StyleValue> Parser::parse_content_value(Vector<ComponentValue> const& com
+            case CSS::ValueID::TableHeaderGroup:
+                return CSS::Display { CSS::Display::Internal::TableHeaderGroup };
+            case CSS::ValueID::TableFooterGroup:
+                return CSS::Display { CSS::Display::Internal::TableFooterGroup };
+            case CSS::ValueID::TableRow:
+                return CSS::Display { CSS::Display::Internal::TableRow };
+            case CSS::ValueID::TableCell:
+                return CSS::Display { CSS::Display::Internal::TableCell };
+            case CSS::ValueID::TableColumnGroup:
+                return CSS::Display { CSS::Display::Internal::TableColumnGroup };
+            case CSS::ValueID::TableColumn:
+                return CSS::Display { CSS::Display::Internal::TableColumn };
+            case CSS::ValueID::TableCaption:
+                return CSS::Display { CSS::Display::Internal::TableCaption };
+            case CSS::ValueID::RubyBase:
+                return CSS::Display { CSS::Display::Internal::RubyBase };
+            case CSS::ValueID::RubyText:
+                return CSS::Display { CSS::Display::Internal::RubyText };
+            case CSS::ValueID::RubyBaseContainer:
+                return CSS::Display { CSS::Display::Internal::RubyBaseContainer };
+            case CSS::ValueID::RubyTextContainer:
+                return CSS::Display { CSS::Display::Internal::RubyTextContainer };
+
+            // display-box
+            case CSS::ValueID::Contents:
+                // FIXME this should be CSS::Display::Box::Contents but contents is currently not implemented
+                return CSS::Display { CSS::Display::Outside::Block, CSS::Display::Inside::Flow };
+            case CSS::ValueID::None:
+                return CSS::Display { CSS::Display::Box::None };
+
+            // display-legacy
+            case CSS::ValueID::InlineBlock:
+                return CSS::Display { CSS::Display::Outside::Inline, CSS::Display::Inside::FlowRoot };
+            case CSS::ValueID::InlineTable:
+                return CSS::Display { CSS::Display::Outside::Inline, CSS::Display::Inside::Table };
+            case CSS::ValueID::InlineFlex:
+                return CSS::Display { CSS::Display::Outside::Inline, CSS::Display::Inside::Flex };
+            case CSS::ValueID::InlineGrid:
+                return CSS::Display { CSS::Display::Outside::Inline, CSS::Display::Inside::Grid };
+
+            default:
+                return {};
+            }
+        }
+        return {};
+    };
+
+    auto parse_double_component_display = [&](Vector<ComponentValue> const& component_values) -> Optional<CSS::Display> {
+        auto first_identifier = parse_identifier_value(component_values.first());
+        auto second_identifier = parse_identifier_value(component_values[1]);
+        if (first_identifier && second_identifier)
+            if (is_display_outside(first_identifier->to_identifier()) && is_display_inside(second_identifier->to_identifier()))
```

Comment:
There's a trap here! When the spec syntax says `||` or `&&` it means that the things can occur in any order. https://www.w3.org/TR/css-values-4/#component-combinators

So, `<display-outside> || <display-inside>` might be `<display-outside> <display-inside>` or  `<display-inside> <display-outside>`. It makes things a lot more fiddly.

Reply:
Ok thank you i thought the positioning is important. I will fix it.